#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_loff_t_dev_t_conflict.dpatch by Jeremie Corbier <jcorbier@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes FTBFS: Conflict between sys/types.h linux/types.h loff_t/dev_t

@DPATCH@
diff -urNad ifplugd-0.28~/src/interface.c ifplugd-0.28/src/interface.c
--- ifplugd-0.28~/src/interface.c	2005-01-04 16:25:31.000000000 -0800
+++ ifplugd-0.28/src/interface.c	2007-01-12 19:19:14.000000000 -0800
@@ -23,7 +23,6 @@
 #endif
 
 #include <linux/sockios.h>
-#include <linux/if_ether.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
diff -urNad ifplugd-0.28~/src/interface.h ifplugd-0.28/src/interface.h
--- ifplugd-0.28~/src/interface.h	2004-05-09 16:20:56.000000000 -0700
+++ ifplugd-0.28/src/interface.h	2007-01-12 19:19:45.000000000 -0800
@@ -21,6 +21,9 @@
  * Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
  */
 
+/* From <linux/if_ether.h> */
+#define ETH_ALEN 6
+
 int interface_auto_up;
 int interface_do_message;
 
